aeon/dj_pipeline/subject.py

```diff
@@ -64,27 +66,45 @@ def make(self, key):
             "eartag": eartag_or_id,
         }
         animal_resp = get_pyrat_data(endpoint=f"animals", params=params)
-        assert len(animal_resp) == 1, f"Found {len(animal_resp)} with eartag {eartag_or_id}, expect one"
-        animal_resp = animal_resp[0]
+        if not animal_resp:
```

Comment:
```suggestion
        if len(animal_resp) == 0:
```

I think this more explicitly filters "no animal" conditions since response may return an error dictionary.